text/0049-protocol-wire-format-negotiation.md

```diff
@@ -0,0 +1,95 @@
+# Protocol wire format negotiation
+
+## Summary
+
+[summary]: #summary
+
+CEP PR: [casperlabs/ceps#0000](https://github.com/casperlabs/ceps/pull/0000)
+
+Allow changing the underlying low-level wire format after handshake.
+
+## Motivation
+
+[motivation]: #motivation
+
+The current low-level wire format of the node has inefficiencies that can be addressed in a backwards compatible manner, improving throughput.
+
+## Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+For backwards compatibility the casper-node has a low-level connection protocol that must not be altered:
+
+* Clients connect to the server using TLS, establishing a bi-directional streaming transport. This is unaffected by this CEP and not discussed further.
+* Client and server send a handshake simultaneously encoded using MessagePack (see below).
+
+Nodes send the following fields in the handshake:
+
+* network name
+* public address
+* protocol version (version 1.1.0 and up)
+* consensus certificate (see [CEP42](0042-network-qos.md))
+
+With the implementation of this CEP, a **fifth** field is added, containing a list of supported wire format versions, an encoded `Set<u8>`, with a default of `[1]`.
```

Comment:
This seems unnecessary since we already exchange protocol versions.  The contents of this proposed fifth field could just be implied by the protocol version couldn't they?

Reply:
Hmm, good point. I will rewrite the CEP (or portions of it) with that aspect in mind.